src/middleware/signed-eth-tx-middleware.ts

```diff
@@ -0,0 +1,109 @@
+import debug from 'debug'
+import { ethers } from 'ethers'
+import { SignedTx, NonceTx, MessageTx, Transaction } from '../proto/loom_pb'
+import { ITxMiddlewareHandler } from '../client'
+import { EthersSigner, soliditySha3 } from '../solidity-helpers'
+import { bytesToHex, hexToBytes } from '../crypto-utils'
+import { Address } from '../address'
+
+const log = debug('signed-eth-tx-middleware')
+
+/**
+ * Signs transactions using Eth crypto primitives
+ */
+export class SignedEthTxMiddleware implements ITxMiddlewareHandler {
+  signer: ethers.Signer
+  fromAddress!: string
+
+  // Apply Eth chain name removes the needing for the account to be mapped
+  applyEthChainName: boolean
```

Comment:
Don't entirely understand what this is for, looks clunky as hell. I'll take a look at the Go code.

Reply:
Ok it's because Piers decided that it should have the chain_name = 'eth' on sign tx when the address mapper exists

Reply:
Can get rid of this field now.